### DISCUSSION TITLE
欧雷的 OpenBuild Web3 前端训练营复盘

### DISCUSSION THREAD

**Discussion by ourai** [openbuildxyz/Web3-Frontend-Bootcamp#1152] [Materials]
哈喽，大家好！👋🏻

在 Web3 前端训练营即将结束之际，正好我也把课后作业几乎都做完了，想借此机会趁热打铁回顾下这一个多月的体验与观察。

很荣幸在学习的过程中被选中参与课后作业审核，不仅让我体会到了开源社区的一些事，还使我有机会提供更多的价值！

## 背景动机

在正式开始复盘之前，我先（重新）自我介绍下——

我叫[欧雷](https://linxoid.com/ourai/)，也可称为「雷叔」，因为已是年过 35 的中年人了，在满是 95 后、00 后的训练营学员中可不就是「叔」了嘛……

自打毕业起就从事 Web 前端开发相关工作，至今已经十多年了；待过多个领域的公司，用过多种技术工具，做过多种类型的项目，担任过多种职业角色——经历还算丰富。

上述职业经历加上近两年的人生经历，令我[持续看衰传统互联网行业 Web 前端开发](https://mp.weixin.qq.com/s/QZZgCydZpsKu6fRu5BByrg)，因而一直在寻找一个能最大化利用自己已有知识与技能的出路。

今年 5 月的某天，忘了为啥我突然想去了解 Web3 相关的东西——在那之前也只是听过，具体是什么基本不知道。

即便如此，想找到进入去中心化世界的入口是我多年以来的愿望，而当下以区块链为基础的 Web3 就是那入口（之一）！

经过一番了解，Web3 的现状与发展趋势比较符合我的想象，尤其是与之相伴而生的 DAO 这种组织形态简直就是我理想中的样子！

从搜到的几个国人 DAO 中选择了与我的理想抱负最为匹配的 [FreeBe](https://free-be.xyz/) 申请加入，之后看到主理人在群中分享的本训练营学员招募文章，所提到的一些收益都/正是我所需要的，于是果断报名！

于我而言，转型为 Web3 全栈开发就是那「最大化利用自己已有知识与技能的出路」——本训练营能够很好地助力！

## 学员视角

虽然中途有调整过，但训练营的课程结构自始至终都大体分为「基础」与「进阶」两大部分：

1. 基础课程——开发一个 dApp 所要掌握的基本要素，并用较常见的项目类型 NFT market 和 DEX 作为实战练习；
2. 进阶课程——推广利益相关方的产品，顺便帮助想要入行的开发者尽早适应目前行业的多变。

### 整体设计

从纯粹的学员角度来看——

训练营的定位更倾向于「帮助有经验的 Web2 前端转型为 Web3 前端」，考虑到会有在读/应届大学生，就安排了一节介绍前端的课程。

但实际报名参加的 200 多人中还有很多或从 Web2 转型或已是 Web3 从业者的后端开发，甚至还有几乎完全零基础的……

由于这是一个免费的训练营，并且举办方还提供了多种奖励设置，不大可能再投入更多资源与成本去塑造得更好，因而：

- 课程内容的体系化、精细度不会像动则成千上万的培训那样完善；
- 课后的辅导等服务也不会像动则成千上万的培训那样体贴细致。

课程视频内容的主要作用就是用提炼出的一些要点在学员脑中画出大致的轮廓，真要学明白的话，还是得课后自己依据那个轮廓进行更深入的扩展与填充。

这时，学员以往的知识结构、思维模式与学习习惯等会显著地影响新内容的学习与吸收速度——有的人很快就掌握了并做好课后作业，有的人则要在群里问好多问题。

就我自身来说，因编程经验丰富且善于思考总结，脑中已经建立了抽象模型网络，大多学起来很容易理解；就算有疑问需要解答，基本问下文心一言就搞定了。

所以，我是经常在训练营群里回答别人问题的人，这不仅解决了他人的困惑，还能验证并巩固我自己对知识的理解。

这样看来，训练营课程弄得「粗糙」非但不是缺点，反而是一石好几鸟的「英明决策」：

1. 举办方以较低的成本攒局，既扩大了他们及利益相关方的影响力，又尽量惠及更多想要转型或学习的人；
2. 互助式学习取代辅导式学习，学得快的帮助学得慢的既能达到双赢效果，又能营造出积极的学习氛围；
3. 间接督促学员学习并完成课后作业，培养从事 Web3 行业的相关习惯。

尽管如此，还是希望他人最好能像看过《[提问的智慧](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way/blob/main/README-zh_CN.md)》一样尽可能提出有价值的问题。

总而言之，我觉得举办方几乎没什么问题，唯一的问题就是担心提交的课后作业能否及时被审核并合进去……

### 学习成果

如果我没学到一些东西的话，那参加训练营的这一个多月时间就是白白浪费了——还好并没！

虽说我是 Web 前端开发出身，也用过 React 去开发应用，但那基本是好多年前了，并且主要用的是 class component，当下主流的 React Hooks 使用经验极少，因而也算是学习「新」技术了。

在几个实战项目的锤炼下，我除了掌握开发出一个完整 dApp 的能力外，还孵化了 3 个学习与教学用 demo 项目：

- [My First To-do List](https://github.com/ourai/my-first-todo-list)——用 GUI 工具开发待办事项列表，现在只有 React Hooks 的实现，接下来学习 Flutter 开发时会再添加个相应的实现；
- [My First NFT Market](https://github.com/ourai/my-first-nft-market)——包含 Solidity 智能合约及前端的完整 dApp，日后将智能合约改为可升级的，并优化前端交互逻辑与用户体验；
- [My First DEX](https://github.com/ourai/my-first-dex)——基于 Uniswap V2 SDK 实现的去中心化交易所，过后找时间再实现个 V3 版本的。

另外，我还产出了《[智能合约纯小白是如何完成自己的第一个 dApp——NFT market？](https://github.com/openbuildxyz/Web3-Frontend-Bootcamp/discussions/821)》与《[百字说清 EVM++ 的本质](https://github.com/openbuildxyz/Web3-Frontend-Bootcamp/discussions/1095)》这两篇学习笔记。

在这个训练营所学到的知识与技能并不会没有用武之处，我正参与的一个 FreeBe 项目中会马上就利用起来！

## 助教视角

可能是因为我在群中表现得很活跃、热心，被叫去帮忙审核课后作业了，这就在「学员」之上又多了个「助教」的身份，立场变得不那么单一、纯粹了。

### 各种不足

成为助教后我知道了，审核人员真的是少得可怜！

第一节 React 课程上线后，报名与提交作业的 PR 铺天盖地袭过来——50、100、150，马上就要到 200 了！可是，人呢？审核 PR 的人呢？！

好了好了，抓了几个壮丁过来，这下人有了，可以开始干活了——咋干？按什么标准去审核？task 1 是只要功能对就行？还是也得审核代码编写情况？

那么多人提交作业，肯定会有人「作弊」直接去复制粘贴别人的代码而不自己动脑做，这样的话还作数吗？若要去管，怎样方便快捷地识别检测出来？

虽然有几点最基本的规则，但还不够清晰、健全；再加上没配置有效的 GitHub Actions 工作流进行自动化处理，更需要大量人力去干涉。

鉴于以上原因，上文中我担心的课后作业审核及时性问题果然出现了！

### 学员画像

以当前时间为准，迄今为止的 1113 个 PR 中有至少 451 个是被我审核过的，主要是报名、task 1、task 7 和 task 8：

<img width="354" alt="reviewed-prs" src="https://github.com/openbuildxyz/Web3-Frontend-Bootcamp/assets/562589/187753c5-81eb-46a0-bea1-5daaa858d3cf">

在这数以千计的 PR 中，估计有一半以上是完全能够避免被创建的；也就是说，在审核人员要求更改时，可以在原 PR 中处理，以保留完整的时间线并减少资源等的浪费。

在审核 PR 过程中，纠正课程内容无关的提交规范、Git 使用方面的问题占用了极多时间，这令我十分纳闷儿——训练营的目标群体到底是哪类人？？？

我本来以为符合以下几点的人会占据绝大多数：

- 不要求学习力、自驱力很强，但至少知道先去看已有的文档与视频；
- 对基本的 Git 操作了如指掌，懂得如何利用分支进行事务隔离。

然而，实际情况却让我大跌眼镜……我都不禁开始为他们担心起来——这能找到 Web3 的工作吗？

在活动接近尾声的现在，我发现有的人是真正来学习的，但也有人看上去就完全是冲着💰来的——稍微需要花点时间的课后作业一个也没做。

我不清楚举办方是否已把这种情形纳入预期，并且不排斥其发生。

## 结语

作为学员，我对这次的训练营活动很是满意，不仅免费且高效地学到了我亟需的知识与技能，更能拿奖励并认识很多人，还要啥自行车啊！

作为助教，我看到了很多问题，除了略微吐槽之外，我也在脑海中想了一些改进方案，希望有机会实施出来解决掉那些问题。

OpenBuild，请再接再厉！

**Comment by keyding** [comment]
同是 Web2 转 Web3 的小学生，互勉 🤝

**Reply by ourai** [reply]
你的个人网站很漂亮！

**Reply by ssrabc** [reply]
你的网站做的可以啊

**Comment by coderhyy** [comment]
其实这些跟风的小白才是大多数😂

**Reply by ourai** [reply]
😑

**Comment by keyding** [comment]
> 同是 Web2 转 Web3 的小学生，互勉 🤝

✌️谢谢～

**Comment by zhengqiuwan** [comment]
**理想version**：我愿称之为一次大规模的去中心化实验（bushi，瞎说🙊），在适当的原则和条件下，大家可以无许可的加入学习探索行动，以此即发更多涌现。自组织的精神下的协作、互信与共创的实验探索，彼此互相学习、互相启发、协同协作，也能够给到更多自我的自由，产生更多创意和行动涌现。这是一直在探索的如何在开源、开放、无需用、无中心的前提下，产生大规模去中心化的社会协作的路径，这也将会帮助创造更好的社区。

**现实version**：实际上确实是人手不足，需要更多社区的小伙伴（幸好有欧雷一起拉磨）➕经验不足，很多规范和规则初步的制定是没有经过实践检验和深度思考，只想着最简流程，至于一些流程上自动化，还在学习！这次从大家（欧雷）身上学到很多。最近很苦恼，有些人会说xx活动不好，xx课程不够厉害。也一直在想什么样的活动/内容是才是好的，才是对大家有价值，没有浪费大家时间的。可有时候好像很难去把控，世界太参差了。还是给自己加个油，先做当下能做的事情，剩下则有则改进，不断优化。

**唯一宗旨：大家通过真正做项目学习！** 相信欧雷是这次收益最大的（按照pow来说毫无疑问一定是激励榜top1）毕竟也是咋们付出最多的学员，审核451 pr 还像奶牛一样疯狂输出文章和笔记，磨盘直接冒火星了！！！感谢欧雷的付出和支持。

**Reply by ssrabc** [reply]
怎么样才能成为审核员
